Helper/AlgoliaHelper.php

```diff
@@ -83,23 +77,25 @@ public function __construct(
 
     /**
      * @return void
-     * @throws AlgoliaException
      */
     protected function createClient(): void
     {
         $storeId = $this->getStoreId();
-        if ($this->algoliaCredentialsManager->checkCredentials($storeId)) {
-            $config = SearchConfig::create(
-                $this->config->getApplicationID($storeId),
-                $this->config->getAPIKey($storeId)
-            );
-            $config->setConnectTimeout($this->config->getConnectionTimeout($storeId));
-            $config->setReadTimeout($this->config->getReadTimeout($storeId));
-            $config->setWriteTimeout($this->config->getWriteTimeout($storeId));
-            $this->clients[$storeId] = SearchClient::createWithConfig($config);
-        } else {
-            throw new AlgoliaException('Client initialization could not be performed because Algolia credentials were not provided.');
+        if (!$this->algoliaCredentialsManager->checkCredentials($storeId)) {
+            if ($storeId !== self::ALGOLIA_DEFAULT_SCOPE) {
```

Comment:
What is the intent behind this logic?

Reply:
@cammonro This is an interesting question. During my tests, I assumed that the user has to fill the credentials in the Default config but with the multi apps approach, I guess they can "forgot" to do it if they fill the credentials at store level. 
The question is, is it still needed now ? Looks like this is not anymore and - I guess - that's ok to let the credentials empty at default config level WDYT ? (cc @mrahman3177 )

Reply:
Thoughts after re-adding the exception, I think we need to require the credentials at default config if we want to make sure this method can't return null 🤔